pkg/validation-webhook/validation-webhook.go

```diff
@@ -0,0 +1,254 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	vckv1alpha1 "github.com/IntelAI/vck/pkg/apis/vck/v1alpha1"
+	"github.com/golang/glog"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+//Config struct
+type Config struct {
+	CertFile string
+	KeyFile  string
+}
+
+var scheme = runtime.NewScheme()
+
+var codecs = serializer.NewCodecFactory(scheme)
+
+type admitFunc func(v1beta1.AdmissionReview) *v1beta1.AdmissionResponse
+
+func toAdmissionResponse(err error) *v1beta1.AdmissionResponse {
+	return &v1beta1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
+func admitVolumeManager(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	glog.V(2).Info("admitting volume manager")
+
+	raw := ar.Request.Object.Raw
+	vm := vckv1alpha1.VolumeManager{}
+	deserializer := codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(raw, nil, &vm); err != nil {
+		glog.Error(err)
+		return toAdmissionResponse(err)
+	}
+
+	reviewResponse := validateVolumeManager(vm)
+
+	return reviewResponse
+}
+
+func validateNFS(vc vckv1alpha1.VolumeConfig) string {
+	errs := []string{}
+	if len(vc.Labels) == 0 {
+		errs = append(errs, "labels cannot be empty.")
+	}
+
+	if _, ok := vc.Options["server"]; !ok {
```

Comment:
Can you also validate the value types?

Reply:
Isn't that done in the OpenAPI schema?

Reply:
no, like for example, the server should be a url, the timeout should be in the right format, etc.